packages/modern-browsers/TODO.md

```diff
@@ -0,0 +1,46 @@
+## Remaining work
+
+- [x] Come up with a system for constraining minimum browser versions.
+
+- [x] Research minimum versions for key ECMAScript features.
+
+- [x] Use the minimum versions in `webapp` to determine JS bundle.
+
+- [x] Import different `core-js` polyfills in `ecmascript-runtime-client`
+      depending on modern/legacy classification.
+
+- [ ] Really vet the set of imported `core-js` polyfills based on known
+      minimum versions imposed by `setMinimumBrowserVersions`.
+
+- [ ] Create an `isobuild:web-browser-legacy` pseudopackage.
+
+- [ ] Add tests to the `modules` test app.
+
+- [x] Expose `Meteor.isModern` on both client and server.
+
+- [ ] Make sure in-browser tests run with both `web.browser` and
+      `web.browser.legacy`.
+
+- [x] Use `web.browser.legacy` to handle `es5-shim` instead.
+
+- [x] Use `web.browser.legacy` to handle SockJS instead.
+
+- [x] Load `SockJS` using dynamic `import()` if necessary in modern
+      `web.browser` clients.
+
+- [ ] Use different plugins in babel-compiler for `web.browser.legacy`.
+
+- [x] Fix dynamic module source map URLs (prepend `/__arch`).
+
+- [x] Fix tests failing because of changes to static resource URLs.
+
+- [ ] In development, save time by only rebuilding `web.browser` (modern)?
+
+- [ ] Try adding a `web.worker` platform and see if it works as expected.
```

Comment:
👀😜 @mitar

Reply:
How hard it would be to allow `web.worker.<tag>`?

Reply:
The main difficulty I see is that any package that adds something to `client` currently would implicitly be adding that resource to `web.worker` as well, since `client` essentially means `web.*`, which would include `web.worker` and/or `web.worker.*`.

That's a problem both because it would make the worker bundles larger and because existing package code might use global browser APIs that aren't available in web workers.

We could call it something different, like `webworker.<tag>`, so it wouldn't inherit all the `web` stuff, but you probably want to have at least some of your packages available in a worker.

Maybe packages have to opt into supporting workers with an `isWorkerSafe: true` option in `Package.describe`?

We also need worker-specific entry points in the application.

Now that I'm thinking about it, there are a lot of unanswered questions here.

Reply:
I would go with `worker.<tag>`. Simple and backwards compatible. Those new packages which want to add things to worker, can do that.

BTW, then we should also add `worker` directory besides `client` and `server` in apps? Or will this work just for packages?

Reply:
One solution is also that if I do `api.use('package', 'worker.web')` and `package` has only `web` available, then it is still pulled in. So if package was meant to be used in the browser but has not been made available for workers, you could pull it in.

Also, is really a bundle still a thing with dynamic imports?

Reply:
> Also, is really a bundle still a thing with dynamic imports?

 If you use `import()`, that just makes multiple bundles, one loaded up front, and some loaded later, right?